From 5983f4d94c508e07ea31f3ce6d2a6e16d6c23439 Mon Sep 17 00:00:00 2001
From: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
Date: Tue, 3 Jul 2018 20:39:22 -0500
Subject: [PATCH 1301/1677] ASoC: SOF: add mm.h

0-day complains that on some architectures such as m68k mm_struct is not
defined. Make sure it's added.

In file included from ./arch/m68k/include/asm/pgtable_mm.h:148,
                 from ./arch/m68k/include/asm/pgtable.h:5,
                 from sound/soc/sof/pm.c:15:
./arch/m68k/include/asm/motorola_pgtable.h: In function 'pgd_offset':
./arch/m68k/include/asm/motorola_pgtable.h:199:11: error: dereferencing pointer to incomplete type 'const struct mm_struct'
  return mm->pgd + pgd_index(address);
           ^~
In file included from ./arch/m68k/include/asm/pgtable_mm.h:148,
                 from ./arch/m68k/include/asm/pgtable.h:5,
                 from sound/soc/sof/core.c:15:
./arch/m68k/include/asm/motorola_pgtable.h: In function 'pgd_offset':
./arch/m68k/include/asm/motorola_pgtable.h:199:11: error: dereferencing pointer to incomplete type 'const struct mm_struct'
  return mm->pgd + pgd_index(address);

Signed-off-by: Pierre-Louis Bossart <pierre-louis.bossart@linux.intel.com>
(cherry picked from commit 0fee76c9ff8cfcdb89bf5967cf14b7dd44a0e15f)
---
 sound/soc/sof/core.c | 1 +
 sound/soc/sof/pm.c   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/sound/soc/sof/core.c b/sound/soc/sof/core.c
index ce31593..de87e8c 100644
--- a/sound/soc/sof/core.c
+++ b/sound/soc/sof/core.c
@@ -8,6 +8,7 @@
  * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
  */
 
+#include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/delay.h>
diff --git a/sound/soc/sof/pm.c b/sound/soc/sof/pm.c
index d825ff4..56c6042 100644
--- a/sound/soc/sof/pm.c
+++ b/sound/soc/sof/pm.c
@@ -8,6 +8,7 @@
  * Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
  */
 
+#include <linux/mm.h>
 #include <linux/module.h>
 #include <linux/slab.h>
 #include <linux/delay.h>
-- 
2.7.4

